File: searching_algorithms/binary_search/binary_search_iterative.py
def binary_search (items, search_item):
    '''ICS iterative version'''
    index = -1
    found = False
    first = 0
    last = len(items) - 1
        
    while first <= last and found == False:
        midpoint = (first + last) // 2
        if items[midpoint] == search_item:
            index = midpoint
            found = True  
        elif items[midpoint] < search_item: 
            first = midpoint + 1 
        else: 
            last = midpoint - 1       
    return index

def test():
    '''test script for binary search'''
    # Create a list containing some names
    items = ['Albie', 'Byron', 'Divya', 'Hari', 'Joanne', 'Kim', 'Wilbur', 'Zoe'] 
    
    # Search for first item in the list
    print ('Searching for first name in the list')
    result = binary_search(items, 'Albie')
    print ('Result of search was {}'.format(result))
    
    # Search for last item in the list
    print ('Searching for the last name in the list')
    result = binary_search(items, 'Zoe')
    print ('Result of search was {}'.format(result))
    
    # Search for an item that is not in the list
    print ('Searching for a name that is not in the list')
    result = binary_search(items, 'Boris')
    print ('Result of search was {}'.format(result))

if __name__ == '__main__':
    test()

File: searching_algorithms/binary_search/binary_search_iterative.cs
using System;

namespace isaac_code_samples
{
    class BinarySearchRecursive
    {
        static void Main(string[] args){
           test(); 
        }

        public static int BinarySearch(int[] items, int search_item){
            // Binary search iterative version
            //Initialise variables
            int index = -1;
            bool found = false;
            int first = 0;
            int last = items.Length - 1;

            // Repeat while there are more items to check and the item has not yet been found.
            while(first <= last && found == false){
                int midpoint = (first + last) / 2;
                if (items[midpoint] == search_item){
                    index = midpoint;
                    found = true;
                }else if (items[midpoint] < search_item){
                    first = midpoint + 1;
                }else{
                    last = midpoint - 1;
                }
            }
            return index;
        }

        private static void test(){
            int[] items = new int[]{5, 12, 24, 56, 68, 72, 81, 95};

            //Search for first item in array
            Console.WriteLine("Searching for first item in array...");
            int result = BinarySearch(items, 5);
            Console.WriteLine("The search result was: " + result.ToString());

            //Search for last item in array
            Console.WriteLine("Searching for last item in array...");
            result = BinarySearch(items, 95);
            Console.WriteLine("The search result was: " + result.ToString());

            //Search for an item that is not in the array
            Console.WriteLine("Searching for last item in array...");
            result = BinarySearch(items, 36);
            Console.WriteLine("The search result was: " + result.ToString());
        }
    }
}